src/batchgetindex.jl

```diff
@@ -110,7 +111,7 @@ function process_index(i::AbstractVector{<:Integer}, cs, ::ChunkRead)
         push!(tempinds, (tempind,))
         maxtempind = max(maxtempind,maximum(tempind))
     end
-    (length(i),), ((maxtempind),), (outinds,), (tempinds,), (datainds,), Base.tail(cs)
+    size(i), ((maxtempind),), (outinds,), (tempinds,), (datainds,), Base.tail(cs)
```

Comment:
This  return tuple is so long, should we make it a `NamedTuple` or even make a struct for it?

Reply:
Yes, in principle it might be better to return a struct here. However, I am quite happy that currently everything infers well and am a bit reluctant to touch this part right now, since these process_index chains are the most demanding parts for the compiler. Could we fix this later?

Reply:
I am preparing another PR that introduces some DiskIndex type, which I will also expand to include some information on aliasing between temp and output array. So I will merge this as is for now